docs/reference-guides/rancher-security/rancher-v2.6-hardening-guides/rke1-self-assessment-guide-with-cis-v1.20-benchmark.md

```diff
@@ -0,0 +1,2762 @@
+```
+
+**Returned Value**:
+
+```console
+root 13371 13350 13 10:28 ? 00:01:08 kube-apiserver --service-account-signing-key-file=/etc/kubernetes/ssl/kube-service-account-token-key.pem --kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname --allow-privileged=true --audit-log-maxbackup=10 --etcd-prefix=/registry --tls-cert-file=/etc/kubernetes/ssl/kube-apiserver.pem --service-account-issuer=rke --anonymous-auth=false --audit-log-format=json --audit-log-maxsize=100 --etcd-certfile=/etc/kubernetes/ssl/kube-node.pem --authentication-token-webhook-cache-ttl=5s --service-account-lookup=true --api-audiences=unknown --proxy-client-cert-file=/etc/kubernetes/ssl/kube-apiserver-proxy-client.pem --proxy-client-key-file=/etc/kubernetes/ssl/kube-apiserver-proxy-client-key.pem --requestheader-username-headers=X-Remote-User --advertise-address=172.31.2.76 --cloud-provider= --etcd-keyfile=/etc/kubernetes/ssl/kube-node-key.pem --etcd-servers=https://172.31.8.188:2379 --kubelet-client-key=/etc/kubernetes/ssl/kube-apiserver-key.pem --secure-port=6443 --requestheader-extra-headers-prefix=X-Remote-Extra- --runtime-config=authorization.k8s.io/v1beta1=true --storage-backend=etcd3 --requestheader-group-headers=X-Remote-Group --profiling=false --service-node-port-range=30000-32767 --audit-log-path=/var/log/kube-audit/audit-log.json --service-account-key-file=/etc/kubernetes/ssl/kube-service-account-token-key.pem --tls-private-key-file=/etc/kubernetes/ssl/kube-apiserver-key.pem --authentication-token-webhook-config-file=/etc/kubernetes/kube-api-authn-webhook.yaml --audit-policy-file=/etc/kubernetes/audit-policy.yaml --audit-log-maxage=30 --client-ca-file=/etc/kubernetes/ssl/kube-ca.pem --kubelet-client-certificate=/etc/kubernetes/ssl/kube-apiserver.pem --bind-address=0.0.0.0 --authorization-mode=Node,RBAC --enable-admission-plugins=NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota,NodeRestriction,Priority,TaintNodesByCondition,PersistentVolumeClaimResize --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305 --insecure-port=0 --requestheader-allowed-names=kube-apiserver-proxy-client --etcd-cafile=/etc/kubernetes/ssl/kube-ca.pem --requestheader-client-ca-file=/etc/kubernetes/ssl/kube-apiserver-requestheader-ca.pem --service-cluster-ip-range=10.43.0.0/16
+```
+
+### 1.2.27 Ensure that the --service-account-key-file argument is set as appropriate (Automated)
+
+
+**Result:** pass
+
+**Remediation:**
+Edit the API server pod specification file /etc/kubernetes/manifests/kube-apiserver.yaml
+on the master node and set the --service-account-key-file parameter
+to the public key file for service accounts:
+--service-account-key-file=<filename>

```

Comment:
```suggestion
`--service-account-key-file=<filename>`

```

Converting to inline code since Docusaurus will throw an error because it treats `<filename>` as an HTML tag.

Reply:
resolved